HW06.Rmd

```diff
@@ -1,92 +1,115 @@
+
+library(magrittr)
+library(knitr)
+library(tidyverse)
+library(tidymodels)
+
+gene_ids <- read.csv("RNA_Seq_processed/WT/1840.sf", sep = "\t", header = TRUE)$Name
+GBM.transcripts <- list.files("RNA_Seq_processed/", full.names = TRUE, recursive = TRUE) %>% 
+  sapply(function(f) {read.csv(f, sep = '\t', header = TRUE)$TPM}) %>% as.data.frame()
+colnames(GBM.transcripts) <- list.files("RNA_Seq_processed/", recursive = TRUE) %>% 
+  strsplit('[\\.]') %>% sapply(function(x) {x[1]})    # Piping can avoid ugly long function nestings
+GBM.transcripts$gene_id <- gene_ids
+
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb, message=FALSE, warning=FALSE}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-gene_id) %>%     # Maybe gene_id is not the first column
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+# Do a bit more: transpose, and separate whether a cell line is WT or mutant
+GBM.genes.of.interest <- ens.mapped_GBM %>%
+  filter(gene_symbol %in% genes_of_interest) %>%
+  pivot_longer(-gene_symbol, names_to = "cell_line", values_to = "TPM") %>%     # tidy up!
```

Comment:
The pivot_longer function is new to me. Looking it up, it seems like a good function for what you wanted here. I will have to keep it in mind for future use!

Reply:
I believe `pivot_longer()` is just an updated version of `gather()`. They do exactly the same things. I think it's good to stick with the new ones.